src/activity_handlers.cpp

```diff
@@ -1022,6 +1005,21 @@ static void butchery_quarter( item *corpse_item, player &p )
 
 void activity_handlers::butcher_finish( player_activity *act, player *p )
 {
+    // No targets means we are done
+    if( act->targets.empty() ) {
+        act->set_to_null();
+        return;
+    }
+
+    item_location &target = act->targets.back();
+
+    // Corpses can disappear (rezzing!), so check for that
+    if( !target || !target->is_corpse() ) {
```

Comment:
The `operator bool` on `item_location` is definitely doing the wrong thing in general.  That's not really your fault, but this can totally end up pointing at the wrong item (I think there have been bug reports about butchery doing that already).  I think the only way to fix this sensibly is with the `weak_ptr` idea Kevin has discussed.  But that's orthogonal to your changes.

Reply:
I think what it's doing is actually OK here for now.  It just searches at its stored location for an item matching its stored pointer, returning true if such an item is found.

I agree the`weak_ptr` plan is definitely a lot cleaner and more robust and may look at implementing that once this get wrapped up.

Reply:
Bear in mind that `colony` deliberately tries to reuse memory, so there is a good chance that there will be some different `item` at the same address if one is removed and another added, and it has no way to tell the difference.

Reply:
Right, but I don't think there's anything we can do about it yet without the `weak_ptr` trick aside from hope it's not a corpse.